src/resolve/resolved-dns-stub.c

```diff
@@ -413,11 +413,17 @@ static int set_dns_stub_common_socket_options(int fd) {
         if (r < 0)
                 return r;
 
-        r = setsockopt_int(fd, IPPROTO_IP, IP_PKTINFO, true);
-        if (r < 0)
-                return r;
+        if (family == AF_INET) {
+                r = setsockopt_int(fd, IPPROTO_IP, IP_PKTINFO, true);
+                if (r < 0)
+                        return r;
 
-        return setsockopt_int(fd, IPPROTO_IP, IP_RECVTTL, true);
+                r = setsockopt_int(fd, IPPROTO_IP, IP_RECVTTL, true);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
hmm, but we should set the matching optoins for ipv6 here, i.e. IPV6_RECVPKTINFO + IPV6_RECVHOPLIMIT